NEXT-20738 - Added documentation for enterprise project setups (#524)
shopware/docs: guides/hosting/installation-updates/enterprise-project-setup.md

# Enterprise project setup

The setup of enterprise systems differs from a normal installation of Shopware. Most enterprise systems are a completely customized store with individual templates and extensions to make the store work the way it should.

> This guide is meant to serve as a suffering guide for all those who do not have any experience in this area yet.

Reviewer: Is "suffering guide" a turn of phrase? It's sounds rather negative. I think guide should suffice.

> This guide is meant to serve as a suffering guide for all those who do not have any experience in this area yet.

Author: ```suggestion
This guide is intended to serve as a guide for those who have no experience in this area.
```

## Production template
We start with the [production](composer#shopware-6-production-template) template. We recommend this repository for use on production systems. A fork should be made of the template, in which the desired Shopware repositories (core, storefront, admin, elastic, etc) are then pinned to the desired version.

By pinning the versions, it is prevented that unwanted updates are imported into the system when deploying.

## Sources
The following folders are available in the production template:
- **/src**: Here the project specific bundles and sources can be stored.
- **/config**: Here are the .yaml config files as well as other possible configurations (routing, admin configs, etc).
- **/tests**: Here you can store the tests for the project specific sources.
- **/config/bundles.php**: In this file all Symfonfy bundles are defined, which should be included in the project.

## Third party sources

> Most enterprise projects are projects in which the development team is responsible for the stability and performance of the system. The integration of external sources via apps or plugins is of course not forbidden, but should always be viewed with a critical eye. By including those sources, the development team partially relinquishes control over the system. If some plugins are necessary, we recommend not to include them as a installable and uninstallable plugin, but as a composer plugin.

Reviewer: I'd say it's more to provide a plugin **through** composer. So it's not misunderstood as something like the composer-bin plugin.

> Most enterprise projects are projects in which the development team is responsible for the stability and performance of the system. The integration of external sources via apps or plugins is of course not forbidden, but should always be viewed with a critical eye. By including those sources, the development team partially relinquishes control over the system. If some plugins are necessary, we recommend not to include them as a installable and uninstallable plugin, but as a composer plugin.

Reviewer: ```suggestion
Most enterprise projects are projects in which the development team is responsible for the stability and performance of the system. The integration of external sources via apps or plugins is of course not forbidden, but should always be viewed with a critical eye. By including those sources, the development team partially relinquishes control over the system. If some plugins are necessary, we recommend not to include them as a installable and uninstallable plugin, but as a package through composer.
```
